### ISSUE TITLE
Write tests for gathering and alginment with compounds

### ISSUE THREAD

**Issue by dcram** [termsuite/termsuite-core#55]

**Comment by dcram** [comment]
Implemented intelligent and exhaustive lemma-lemma pairs event with mwt having multiple compound swts.

See the example in the specs:

``` java
        Term complexTerm = termFactory.create("N:abcdef|abcdef", "N:simple|simple", "P:de|de", "N:xyzt|xyzt");

        assertThat(provider.getClasses(termIndex, termWithSize2Comp))
            .hasSize(1).contains(
                "xy+zt");

        assertThat(provider.getClasses(termIndex, termWithSize3Comp))
            .hasSize(5).contains(
                    "ab+cdef", "abcd+ef",
                    "ab+cd", "ab+ef", "cd+ef"
                );

        assertThat(provider.getClasses(termIndex, complexTerm))
            .hasSize(33).contains(
                    /* inner classes */
                    "ab+cd","ab+ef","ab+cdef",
                    "abcd+ef", "cd+ef",
                    "xy+zt",
                    "ab+simple", "ab+xy", "ab+zt", "ab+xyzt",
                    "cd+simple", "cd+xy", "cd+zt", "cd+xyzt",
                    "ef+simple", "ef+xy", "ef+zt", "ef+xyzt",
                    "abcd+simple", "abcd+xy", "abcd+zt", "abcd+xyzt",
                    "cdef+simple", "cdef+xy", "cdef+zt", "cdef+xyzt",
                    "abcdef+simple", "abcdef+xy", "abcdef+zt", "abcdef+xyzt",
                    "simple+xy", "simple+zt", "simple+xyzt"
                );

```

**Comment by dcram** [comment]
Merged into 2.2 (but tests for alignments not implemented)
